### PR TITLE
Security policy

### PR DESCRIPTION
**Pull request by hamnis**
Added a first draft of a security policy.
Please critique as much as possible. 

All maintainers that want to be added to the list should let me know, or add themselves in the PR.

We should probably say something about announcing the security release. A security advisory is not enough. Maybe someone with a bit more info on how things are done can help with this.

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+| ------- | ------------------ |
+| 0.21.x   | :white_check_mark: |
+| 0.20.x   | :white_check_mark: |
+| 0.19.x   | :x: |
+| 0.18.x   | :x: |
+| < 0.18   | :x: |
+
+## Reporting a Vulnerability
+
+We will use [keybase](https://keybase.io) as the vehicle for reporting security issues as that gives us a
```
Comment (kevinmeredith) [inline SECURITY.md]: Why, I genuinely ask, do you prefer to use Keybase over just opening a new Github issue?

Comment (hamnis) [reply SECURITY.md]: Discussing security severe security problems in the open allows attackers to see that as well. Once there is a security patch there is usually a race between threat actors to exploit and users to upgrade. We want users to be able to upgrade before they become attacked.

Comment (hamnis) [reply SECURITY.md]: AFAIK there is no way to limit a github issue to just be visible to the reporter and the maintainers. That would be ideal. Using the tools that we already know and use day to day.

Comment (kevinmeredith) [reply SECURITY.md]: Thanks, @hamnis, for the explanation.

Comment (rossabaker) [reply SECURITY.md]: GitHub's Security feature gives us a private fork, which is helpful for sharing code and discussing the solution among project owners and invited parties.  I'm unclear whether the security advisory exposes draft-status comments to the public.  We're learning about this tool on the fly, but I think Keybase is useful at minimum for initial disclosure.

File: SECURITY.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+much better feedback loop. We want to work with the reporter to make sure we understand the reported issue.
+
+If the vulnerability is found to be true we will release a patch version of our software
```
Comment (rossabaker) [inline SECURITY.md]: Vulnerability as a boolean is awkward phrasing.

```suggestion
If the issue is deemed a vulnerability, will release a patch version of our software
```

File: SECURITY.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+and make sure that finds it way to Maven Central before we push the patch to github.
```
Comment (rossabaker) [inline SECURITY.md]: Is it worth mentioning that source jars are still available for every binary on Maven Central?

File: SECURITY.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+| 0.21.x   | :white_check_mark: |
+| 0.20.x   | :white_check_mark: |
+| 0.19.x   | :x: |
+| 0.18.x   | :x: |
+| < 0.18   | :x: |
+
+## Reporting a Vulnerability
+
+We will use [keybase](https://keybase.io) as the vehicle for reporting security issues as that gives us a
+much better feedback loop. We want to work with the reporter to make sure we understand the reported issue.
```
Comment (rossabaker) [inline SECURITY.md]:
```suggestion
forum to discuss, analyze, and remediate the threat before an exploit is published.
[Responsible disclosure](https://en.wikipedia.org/wiki/Responsible_disclosure) enhances security for the entire community.
```

File: SECURITY.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+If the vulnerability is found to be true we will release a patch version of our software
+and make sure that finds it way to Maven Central before we push the patch to github.
+After the patch is available on Maven Central, we will also provide a security advisory through github.
```
Comment (rossabaker) [inline SECURITY.md]: We can link:

```suggestion
After the patch is available on Maven Central, we will also provide a [security advisory](https://github.com/http4s/http4s/security/advisories) through github.
```

File: SECURITY.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+| 0.18.x   | :x: |
+| < 0.18   | :x: |
+
+## Reporting a Vulnerability
+
+We will use [keybase](https://keybase.io) as the vehicle for reporting security issues as that gives us a
+forum to discuss, analyze, and remediate the threat before an exploit is published.
+[Responsible disclosure](https://en.wikipedia.org/wiki/Responsible_disclosure) enhances security for the entire community.
+
+If issue is deemed a vulnerability, we will release a patch version of our software
```
Comment (aeons) [inline SECURITY.md]:
```suggestion
If the issue is deemed a vulnerability, we will release a patch version of our software
```

**Pull request merged**